go/pkg/client/cas_upload.go

```diff
@@ -246,9 +246,28 @@ func (c *Client) uploadUnified(ctx context.Context, entries ...*uploadinfo.Entry
 		return nil, 0, err
 	}
 	wait := make(chan *uploadResponse, uploads)
-	var missing []digest.Digest
+	var dgs []digest.Digest
+	dedupDgs := make(map[digest.Digest]bool, uploads)
+	for _, ue := range entries {
+		if _, ok := dedupDgs[ue.Digest]; !ok {
+			dgs = append(dgs, ue.Digest)
+			dedupDgs[ue.Digest] = true
+		}
+	}
+	missing, err := c.MissingBlobs(ctx, dgs)
+	if err != nil {
+		return nil, 0, err
+	}
+	missingDgs := make(map[digest.Digest]bool)
+	for _, dg := range missing {
+		missingDgs[dg] = true
+	}
 	var reqs []*uploadRequest
 	for _, ue := range entries {
+		if _, ok := missingDgs[ue.Digest]; !ok {
+			uploads--
```

Comment:
It looks like this counter is redundant. The receiver loop below can be guarded using `len(reqs)` since that is effectively the number of sent requests and consequently the number of expected responses.

Reply:
Good catch. Done.